src/snowflake/snowpark/dataframe_writer.py

```diff
@@ -53,11 +76,18 @@ class DataFrameWriter:
        specified destination.
     """
 
-    def __init__(self, dataframe: "snowflake.snowpark.dataframe.DataFrame") -> None:
+    @publicapi
+    def __init__(
+        self,
+        dataframe: "snowflake.snowpark.dataframe.DataFrame",
+        _ast_stmt: Optional[proto.Assign] = None,
+    ) -> None:
         self._dataframe = dataframe
         self._save_mode = SaveMode.ERROR_IF_EXISTS
+        self._ast_stmt = _ast_stmt
 
-    def mode(self, save_mode: str) -> "DataFrameWriter":
+    @publicapi
+    def mode(self, save_mode: str, _emit_ast: bool = True) -> "DataFrameWriter":
         """Set the save mode of this :class:`DataFrameWriter`.
```

Comment:
FYI, some more public APIs were added - `.partition_by`, `option`, `options`

Reply:
@sfc-gh-oplaton is right now working on refreshing with main. I added a TODO with a Jira ticket to add these once we refreshed, thanks for the heads up!